### ISSUE TITLE
The positions from #scene.pickFromRay not on model surface

### ISSUE THREAD

**Issue by criyson** [CesiumGS/cesium#8439]
Sandcastle example:  [link](https://sandcastle.cesium.com/#c=pVZbbxtFFP4ro6goa7odr+1dX3KDNHHaSqGJHFNAcYTG3rE97XjHmpmNcSO/gQQPPPJWIfUP9IEHhISAPwMp/AvO3ncchxf2wd6dc75z/ebMXBOJrhldUIn2UUAX6IgqFs7wi3jN2h7Fn0ci0IQFVG7b6GYQIHjUlPhiodAO0jKkg2BV2R0EgyCxhdWIBhRPuBhS7NO5nvap0ocTsKF0n0oJL+AvQsYowEEcmnGqqDYDSf4ax/1EaKXuQ8l30DbG1Qsym3N6TDSpGqrVFJC/VFPz+KUSwXYecLaq9JLT+11fROLMuU/HUIwo93QBIcLnUwIL2w9umL9CH6A62t/fRw76CDnYA0ENO9uJ8spO/keCCxlB5GRIrLrn2aj4eXATW1xV7kYqKfGX51LMmKJYT2lgjcNgpJkIrFSlksVldGMu2Yxpdk0VJr6f6+4aqq+FmPVFJrTL5XgKflkwOWd6NO2RYEItBzs2egT52Vnv8FCEQaR1MZ9SSbEESKjQh6iOnUrkalXBAmKWCwi+iBsYIWQe9UgESnCKuZikkt28CINgwQIgHgZnfTajItSFmdyCBrZZMQrK6ThOgs30UnFeJKCe0kTqcwH0BApk7YcleCNBA4+h2sd0IilVh0De5VPKJlOtrMus/Qg9anm4WXM6zabXabba7aZrIxcK5Nbqjue6rtdoNTs28pwEcpWElPmngX8uFIui+58RuE7D8zo1r912m2kEntPouE4LoqvVowhsdBfUclsNt+k13UYOqjutWrvZajhuDLqDqXk18FHvdFr1FON22jXw1Gi2250S5irjWJFvNAdIAMMFkr28MsRjIa1YBUXbZxf+94zyYE6DiZ6C4OHDSrEBEZqz0asTKFOPLK2in5fOlW3gL1keT/TkkeB5qKZWIYieEvufUDGjWi6fpfrWjakaPZNUBzZ1CXku+JLDvMgsbELG8ecE2EGX/50AurI321gwX0+TaXOPxjUFTn11IuSMaFBci/EomkmH8zklMkoSv+j2+t3Pvzw5631y2L9rcFXZ4IVoLdkw1OaANJ9s9mVEjz7XK3yY2YnJH6tYZX18dNZ7fnJ69lm39/j0027FrlYTs+ivP7755+2vf7/7/f1v79Cfv3z//uufb396e/vmu9sfvr398c2GPMylVaX4zriyWqdwPk7NQ+M8Wy63ebKWWlSZnHilEpK88psptNaedSZpWFU8hGkftXZMuKKFQpbUam3ibzwc8s94iJZGp5qKxbNAU6lovKAsSVXItTFNxfAlSKEuiQwn36lbNkZWqpCVQIw3HbknlOhQ0vIeT3A4afO+wZ6YJIfRmWmS5IvuKVDENqA4PlvXOpuWA2oHiadlKE+XdPtBYdOsspVS/yjnbK4E89eZH52DzOxpMdzhHIWpm/2YWwp2KZWM8LW9gp/0ut3npeaWersyzrqNU1Hb8UCBF6NvPpNJVzceQUE0Mzh7Ta27MhUOgXwjbWV2bVRydk/alcr9kt0iKjjtzC22lkoRtoGKmwRAk+alaoBdGw6fSomXqYPkfudn1DZOmXs3QMGmuAFb9tZefKc8SNY/ZrO5kDq6uFpwbdUUrq3QW1UdhqNXcHEaKRWZ2KtmoD2fXSPm7w+21u7ggy004kQpkIxDzi+gI4Otg70q6BswLuL72hlMfE6Wkcq0dnCaLGKM96rweRelheBDIksW/wU
)

Browser: Chrome 78.0.3904.108

Operating System: mac OS

<img width="433" alt="WX20191205-235541@2x" src="https://user-images.githubusercontent.com/12904924/70251689-eb76aa80-17ba-11ea-8766-b265b4b1b762.png">

**Sometimes I can get the right intersection with ray and model, but most of time get the position is inside the model (Insufficient accuracy).**

**Comment by lilleyse** [comment]
Hm this seems to be hardware dependent. I see inaccurate positions like this on my 2015 Macbook with the AMD graphics card but not with the Intel integrated graphics nor on my linux desktop with an NVIDIA card.

**Comment by liyangis** [comment]
position outside the 3Dtiles model
[link](https://sandcastle.cesium.com/#c=nVdLb9tGEP4rrC+mUGbFt0jHCerYcmLAjg1FSRpYRrEWV9I2FFfYXUpRC91aoD302FtQIH8ghx6KAkXbP9M67b/o8E1KcoOEeYi7O+/5ZnY4x1yZU7IgXLmnRGShHBJB4yl6lu6pu8N0ecgiiWlE+K6mfD2IFHgk4Ry2Ljib0wC49wrOISdYkueMh0E/o1Fbg2jVujuI5qBMDElEQFemFKVLOBpEdKSon6RLNAzxdNZnjwgdT+STeDZjXJKgVahe0ChgC4RDwqW6259QoVxzthBgRcCIUCImFZFxKQ1RZ0zIIwKehCRAu4lFq0RzYpaEPUEkGJaZMON0SiWdE4FwEKiZ4lp8sh/rqJ/xqblpyRPzsAzGCYt6RLCYDwkacTY9EEB8Eqi27rluK+NZwW8rDUHuPZ4SjhEQJkkoJAdESBphSVm01zAEQ2wExZGlGqbh2oaB9I7tdTq+6dqacsfuWK5lWMiA3Y7jWZoCug3L8ZFjmI7jGh3TbmmZDsYpgURv6HhEcECj8QWVw0mPhaFqI9PyfdMxdMtxPcv1QJGObN8wTN/yLNvTTVCmKS4yO75n+6Zv6p7nuKUmEgV9jiMxYnxaQecMS05f2ejkqPu4f9J/kcNmED3BUTDEQoYkyUafsfAa8wexlCxSd8eQNhoBHskwsVwARkdxlL6rrQquQpY4TP4UJPlJQZdCVEJMLxiNBMBhI8xpHo/ImBMiDgDfywxbQr2sIHCn4yDHd3XXNz3TtCzH1mwd6ZbXMQ1ftyAevqv5PgIUWI7pWp5rwGvGf5VZWBgDkbpggqaefaQ5mTUd3TJsswMQcZ3MGt+Bfx6Aw4NXzfeQmSDFMTuuY9o+5HTdofeJcDdEfKgN+nYBaUySlyoukB2JI2hPEJTLq8YxgEpNSeBIV+7C734jjCgk0VhO4ODTT8vEJ8+MDl8eQzh7eKlWILjUr7QG/yUt7Ume0hI0i8VErQ7WWsZDwqZE8uVJTl/vGcUzzmmURv1dsHAZQvstJGzjTO0vgbKnXP6/A8qVtl3GggZyAvzQRm6hmEPbJa+OoXKxrGq3sPGQhYwfzGYE88RJ9Kzb63c//+L4vHd20N8UuGpt0YIl9IHrGBAO8m/xdZjoqdSnatcjfFDISYskJVHr9OjwvPf4+PT8ebf34PRpt6W125lY5e8/v/33zW//vP3j3e9vlb9+/eHdN7/c/Pzm5vX3Nz9+d/PT6y1+NLdWrWpdw0odpSVHfhNC54Xs5PdNHZR5YLcjguPZhA5FExElDvY+rn19fAvTtgn54M5TF3LVBEgKzz3FbGy22wrmw/5yRkqHD7I16j16evagQQyoJZzisAkd9LDX7T6u6KoEror8JS2lHAua09JFsV3Pw3gNjQmYy15RMwmXxbK96tcqar34ZXKPhjGMDkk1jnAoyKYblRO1sWt9yCmX+WRUXpFiwhYnyQ0r8itW5UTEoWzcmuz6SziFuGRnKFvnapPhLicoQsBG22apYxgeY07qbTnjQ1ll3mtmLZ2pwtkEN+v6RfcUqlprsMK8CHRFHFaNcNxee1khQWBzr4qdWv5IGNKZYDRYb1Ycpiaq3Dat6QhmseK/1laIKu/BaC23q8ZMs/Uik1p6B8BLI28BzQenrdNFlLT5kH5F1M0zEV8D+IZSLeRqSk3ZLW63Wref1AoNGlOzxNZcqcxucKVJWvu6QPVogFwN5oVWDZe5goCMoNiCAtqNweDWAqjQBH93tJ19IZchuZ/tfkan6TcIfBCoCLUlmc5CyKxoX8fDlwRgKUQiYL9dMO0HdK7Q4N5gZ+2za7CTfMgIASejOAyfQD4GO/f320DfYAtZOqifwxUd4mVCMjHun2abCKH9Niw3uWQ2T9ck/gc)
![image](https://user-images.githubusercontent.com/34089375/77500025-ca7fb580-6e8e-11ea-8840-2754051fd0c7.png)

**Comment by liyangis** [comment]
![image](https://user-images.githubusercontent.com/34089375/77500160-35c98780-6e8f-11ea-8b1a-5e0bc98e2b0c.png)

**Comment by lilleyse** [comment]
@liyangis what GPU do you have? The accuracy issues seem to be hardware dependent.

**Comment by liyangis** [comment]
> @liyangis what GPU do you have? The accuracy issues seem to be hardware dependent.

![image](https://user-images.githubusercontent.com/34089375/77501217-534c2080-6e92-11ea-8f41-5f76eebc2e62.png)

**Comment by liyangis** [comment]
![image](https://user-images.githubusercontent.com/34089375/77501821-12550b80-6e94-11ea-9793-cf1d2406e8e5.png)

**Comment by liyangis** [comment]
![image](https://user-images.githubusercontent.com/34089375/77501861-2d278000-6e94-11ea-938f-fbac99a87c56.png)

**Comment by dennisadams** [comment]
I managed to (sometimes) reproduce the inaccuracies in the examples of @criyson and @liyangis on my linux desktop.
It appears that the inaccuracies occur when the pick's orthographic frustum bounds are retained from the scene's prespective frustum bounds.
In @criyson's example, when the pick is accurate, the frustum splits are `[0.1, 100, 2265.88...]` (the last plane might vary, depending on resolution, due to SSE).  
When it is inaccurate, the frustum splits are `[0.1, 100000000, 100000000000000000]`, which is based on the `logarithmicDepthFarToNearRatio` (while it should use the `farToNearRatio` like in the previous case). 
So I believe that in the second case the inaccuracies are caused by precision issues (possibly hardware dependent). 

My attempted workaround was to force a frustum update by setting `scene.view.updateFrustums = true` in `Picking.js`'s `getRayIntersection`. 
But luckily this is unnecessary since I've found out that [#8850](https://github.com/CesiumGS/cesium/pull/8850) fixes the issue. 
As @IanLilleyT wrote over there:
> Problem is, the heuristics it was using to decide if the frustums needed to be adjusted was not triggered when switching to orthographic camera, resulting in the orthographic transform using near / far bounds meant for logarithmic depth (since perspective camera uses logdepth and we just switched away from it). This resulted in an extremely tiny value in the orthographic matrix which ended up becoming 0 by the time it was converted to float32 for the shader.
